service/history/workflowTaskHandler.go

```diff
@@ -260,7 +265,30 @@ func (handler *workflowTaskHandlerImpl) handleMessages(
 	return nil
 }
 
-func (handler *workflowTaskHandlerImpl) handleMessage(_ context.Context, _ *protocolpb.Message) error {
+func (handler *workflowTaskHandlerImpl) handleMessage(ctx context.Context, message *protocolpb.Message) error {
+
+	if types.Is(message.GetBody(), (*updatepb.Acceptance)(nil)) {
+		var acceptance updatepb.Acceptance
+		err := types.UnmarshalAny(message.GetBody(), &acceptance)
+		if err != nil {
+			return err
+		}
+		return handler.handleMessageAcceptWorkflowExecutionUpdate(ctx, message.GetProtocolInstanceId(), &acceptance)
+	} else if types.Is(message.GetBody(), (*updatepb.Response)(nil)) {
+		var response updatepb.Response
+		err := types.UnmarshalAny(message.GetBody(), &response)
+		if err != nil {
+			return err
+		}
+		return handler.handleMessageCompleteWorkflowExecutionUpdate(ctx, message.GetProtocolInstanceId(), &response)
+	} else if types.Is(message.GetBody(), (*updatepb.Rejection)(nil)) {
+		var rejection updatepb.Rejection
+		err := types.UnmarshalAny(message.GetBody(), &rejection)
+		if err != nil {
+			return err
+		}
+		return handler.handleMessageRejectWorkflowExecutionUpdate(ctx, message.GetProtocolInstanceId(), &rejection)
+	}
 
 	return nil
```

Comment:
return error for unknown messages?

Reply:
There is such check in `validateMessages`. I intentionally don't want to duplicate that error here. We haven't decided yet, but most likely unknown messages should be just ignored. There might be a way to inject user message handler for user message types. This is why we use `Any` instead of `oneof` here.